### DISCUSSION TITLE
Created a custom server with express + Next.js, entire page reloads when route to custom route path

### DISCUSSION THREAD

**Discussion by sharunthomas** [vercel/next.js#15927] [Help]
# Bug report

## Describe the bug
Below is my server.js file, I created two pages a.js and b.js and routed with URL  `/atest`  and  `/btest`
When I try to load the page a using URL ` /atest` using  `Link`  Next client tries to load atest.js and gives error then the entire app re-renders which is not an acceptable behaviour.
The next client attempts to use a PageLoader to pull in the chunk for the file-named pathname at runtime.

```

const express = require('express')
const next = require('next')

const port = parseInt(process.env.PORT, 10) || 3000
const dev = process.env.NODE_ENV !== 'production'
const app = next({ dev , conf:{useFileSystemPublicRoutes: false}})
const handle = app.getRequestHandler()

app.prepare().then(() => {
  const server = express()

  server.get('/atest', (req, res) => {
    return app.render(req, res, '/a', req.query)
  })

  server.get('/btest', (req, res) => {
    return app.render(req, res, '/b', req.query)
  })

  server.all('*', (req, res) => {
    return handle(req, res)
  })

  server.listen(port, (err) => {
    if (err) throw err
    console.log(`> Ready on http://localhost:${port}`)
  })
})
```
## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1.git clone https://github.com/sharunthomas/custom-server-next-prototype.git
2. `npm run dev` and visit localhost:3000
3. Click on Gallery menu link
4. Observe the entire page reloads.
5. Observe the network you can see a call for atest.js will take place.

## Expected behavior
The call for atest.js should not take place.
The entire Page should not reload, but instead page transition should happen client-side.

## Screenshots

N/A

## System information

- OS: Windows
- Browser: chrome
- Version of Next.js: 9.4.4
- Version of Node.js: 12.18.2

## Additional context

N/A

**Comment by will-stone** [comment]
Hi. Did you ever find a solution for this?
